code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -449,29 +461,23 @@
 		if(INTENT_HELP, INTENT_GRAB, INTENT_DISARM) //Disperse
 			for(var/i in 1 to illusion_count)
 				illusions += new /mob/illusion/xeno(owner.loc, owner, null, illusion_life_time)
-	addtimer(CALLBACK(src, .proc/clean_illusions), illusion_life_time)
+
 
 /// Clean up the illusions list
 /datum/action/xeno_action/mirage/proc/clean_illusions()
 	illusions = list()
+	illusions_active = FALSE
+	if (action_cooldown_check())
+		add_cooldown()
 
-/datum/action/xeno_action/swap
-	name = "Swap"
-	action_icon_state = "hyperposition"
-	mechanics_text = "Swap our position with an illusion."
-	plasma_cost = 50
-	keybind_signal = COMSIG_XENOABILITY_SWAP
-	cooldown_timer = 40 SECONDS
-
-/datum/action/xeno_action/swap/action_activate()
-	succeed_activate()
-	add_cooldown()
+/// Swap places of hunter and an illusion
+/datum/action/xeno_action/mirage/proc/swap()
 	var/mob/living/carbon/xenomorph/X = owner
 	X.playsound_local(X, 'sound/effects/swap.ogg', 10, 0, 1)
 	var/datum/action/xeno_action/mirage/mirage_action = X.actions_by_path[/datum/action/xeno_action/mirage]
 	if(!mirage_action?.illusions)
 		to_chat(X, span_xenowarning("We have no illusions to swap with!"))
-		return FALSE
```

Comment:
why did this FALSE return exist ?

Reply:
Old return value on action/xeno_action/swap/action_activate(). As far as I can see action_activate()'s return isn't used anywhere, and the overrides in the codebase are inconsistent about having a return value. So, I didn't include the return value in action/xeno_action/mirage/action_activate().